File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& v, int x) {
        int sum=accumulate(v.begin(),v.end(),0);
       
        
        int target=sum-x;
        
        if(target==sum) return v.size();
        if(target<0) return -1;
        sum=0;
        int n=v.size();
        int l=0;
        int ans=-1;
        for(int i=0;i<n;i++)
        {
            sum+=v[i];
            
            while(sum>target)
            {
                sum-=v[l];
                l++;
            }
            
            if(sum==target)
            {
                ans=max(ans,i-l+1);
            }
           
        }
        
        return ans==-1?-1:n-ans;
        
    }
};

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        
        s=sum(nums)
        if s<x:
            return -1
        n=len(nums)
        target=s-x
        s=0
        
        l=0
        ans=-1
        for i in range(len(nums)):
            s+=nums[i]
            
            
            while s>target:
                s-=nums[l]
                l+=1
            if s==target:
                ans=max(ans,i-l+1)
                
        return n-ans if ans!=-1 else -1